src/sentry/issues/priority.py

```diff
@@ -93,32 +97,28 @@ def get_priority_for_ongoing_group(group: Group) -> PriorityLevel | None:
     if not features.has("projects:issue-priority", group.project, actor=None):
         return None
 
-    previous_priority_history = (
-        GroupHistory.objects.filter(
-            group_id=group.id, status__in=PRIORITY_TO_GROUP_HISTORY_STATUS.values()
-        )
-        .order_by("-date_added")
-        .first()
-    )
+    previous_priority_history = GroupHistory.objects.filter(
+        group_id=group.id, status__in=PRIORITY_TO_GROUP_HISTORY_STATUS.values()
+    ).order_by("-date_added")
 
-    new_priority = (
-        [
-            priority
-            for priority, status in PRIORITY_TO_GROUP_HISTORY_STATUS.items()
-            if status == previous_priority_history.status
-        ][0]
-        if previous_priority_history
-        else get_initial_priority(group)
-    )
+    new_priority = None
+    if len(previous_priority_history) > 1:
+        # Skip the most recent history entry since that is the current state
+        previous_history = previous_priority_history[1].status
+        new_priority = GROUP_HISTORY_STATUS_TO_PRIORITY.get(previous_history, None)
+
+    if not new_priority:
+        # If we can't find a previous priority history entry, fall back to the initial priority
```

Comment:
I think you might be able to remove the above logic completely and only use this initial priority logic if you wanted. Since priority gets locked when users change it, the previous priority value should always be the initial one.

Reply:
big brain 😮